### PR TITLE
Add `scope` column to `duckdb_settings`

### PR DESCRIPTION
**Pull request by Tishj**
This allows us to differentiate between global (dbconfig) and local (clientconfig) settings.
~~Making this as a draft because it likely requires patching on all extensions that make use of TryGetCurrentSetting~~
Thanks Carlo, this way no changes in extensions are required!

### CODE REVIEW

File: src/function/table/system/duckdb_settings.cpp [hunk 1]
```diff
@@ -50,20 +54,23 @@ unique_ptr<GlobalTableFunctionState> DuckDBSettingsInit(ClientContext &context,
 		value.value = option->get_setting(context).ToString();
 		value.description = option->description;
 		value.input_type = EnumUtil::ToString(option->parameter_type);
+		value.scope = option->set_global ? "GLOBAL" : "LOCAL";
```
Comment (Mause) [inline src/function/table/system/duckdb_settings.cpp:57]: What about settings that can be set both globally and locally?

Comment (Tishj) [reply src/function/table/system/duckdb_settings.cpp:57]: Yea I thought about that, but when I looked at them it looked like it just set the option globally

Comment (Tishj) [reply src/function/table/system/duckdb_settings.cpp:57]:
```c++
//===--------------------------------------------------------------------===//
// Default Collation
//===--------------------------------------------------------------------===//
void DefaultCollationSetting::SetGlobal(DatabaseInstance *db, DBConfig &config, const Value &input) {
	auto parameter = StringUtil::Lower(input.ToString());
	config.options.collation = parameter;
}

void DefaultCollationSetting::ResetGlobal(DatabaseInstance *db, DBConfig &config) {
	config.options.collation = DBConfig().options.collation;
}

void DefaultCollationSetting::ResetLocal(ClientContext &context) {
	auto &config = DBConfig::GetConfig(context);
	config.options.collation = DBConfig().options.collation;
}

void DefaultCollationSetting::SetLocal(ClientContext &context, const Value &input) {
	auto parameter = input.ToString();
	// bind the collation to verify that it exists
	ExpressionBinder::TestCollation(context, parameter);
	auto &config = DBConfig::GetConfig(context);
	config.options.collation = parameter;
}

Value DefaultCollationSetting::GetSetting(ClientContext &context) {
	auto &config = DBConfig::GetConfig(context);
	return Value(config.options.collation);
}
```

**Comment by Tishj** [comment]
One result of this is that extension settings when they're not set result in the INVALID scope
```
┌──────────────────────┬─────────┬──────────────────────────────────────────────────────────────────────────────────────────────────────────────┬────────────┬─────────┐
│         name         │  value  │                                                 description                                                  │ input_type │  scope  │
│       varchar        │ varchar │                                                   varchar                                                    │  varchar   │ varchar │
├──────────────────────┼─────────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────────┼────────────┼─────────┤
│ azure_http_proxy     │         │ Proxy to use when login & performing request to azure. By default it will use the HTTP_PROXY environment v…  │ VARCHAR    │ INVALID │
│ azure_account_name   │         │ Azure account name, when set, the extension will attempt to automatically detect credentials                 │ VARCHAR    │ INVALID │
│ s3_endpoint          │         │ S3 Endpoint (empty for default endpoint)                                                                     │ VARCHAR    │ INVALID │
│ azure_storage_conn…  │         │ Azure connection string, used for authenticating and configuring azure requests                              │ VARCHAR    │ INVALID │
│ s3_access_key_id     │         │ S3 Access Key ID                                                                                             │ VARCHAR    │ INVALID │
│ binary_as_string     │         │ In Parquet files, interpret binary data as a string.                                                         │ BOOLEAN    │ INVALID │
│ azure_proxy_user_n…  │         │ Http proxy user name if needed.                                                                              │ VARCHAR    │ INVALID │
│ s3_session_token     │         │ S3 Session Token                                                                                             │ VARCHAR    │ INVALID │
│ sqlite_all_varchar   │         │ Load all SQLite columns as VARCHAR columns                                                                   │ BOOLEAN    │ INVALID │
│ s3_secret_access_key │         │ S3 Access Key                                                                                                │ VARCHAR    │ INVALID │
│ azure_proxy_password │         │ Http proxy password if needed.                                                                               │ VARCHAR    │ INVALID │
├──────────────────────┴─────────┴──────────────────────────────────────────────────────────────────────────────────────────────────────────────┴────────────┴─────────┤
│ 11 rows                                                                                                                                                    5 columns │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
```

**Comment by carlopi** [comment]
One question: should this require changes in extension code? Could/should that be avoided?

I was thinking that having something like:
```
class SettingLookupResult {
public:
      operator bool() {
            return scope != SettingScope::INVALID;
      }
      //....
private:
      SettingScope scope = SettingScope::INVALID;
};

DUCKDB_API SettingLookupResult TryGetCurrentSetting(const string &key, Value &value);
```
should work (note the implicit conversion to bool) in all callsites.

**Comment by Tishj** [comment]
> One question: should this require changes in extension code? Could/should that be avoided?
> 
> I was thinking that having something like:
> 
> ```c++
> class SettingLookupResult {
> public:
>       operator bool() {
>             return scope != SettingScope::INVALID;
>       }
>       //....
> private:
>       SettingScope scope = SettingScope::INVALID;
> };
> 
> DUCKDB_API SettingLookupResult TryGetCurrentSetting(const string &key, Value &value);
> ```
> 
> should work (note the implicit conversion to bool) in all callsites.

Hmm I hadn't considered that approach, that might work?

**Comment by Mytherin** [comment]
Thanks!

**Pull request merged**